modules/tinymce/src/core/main/ts/api/UndoManager.ts

```diff
@@ -50,6 +51,29 @@ const UndoManager = (editor: Editor): UndoManager => {
       return Rtc.addUndoLevel(editor, undoManager, index, locks, beforeBookmark, level, event);
     },
 
+    /**
+     * Dispatch a change event with current editor status as level and current undoManager layer as lastLevel
+     *
+     * @method dispatchChange
+     */
+    dispatchChange: (): void => {
+      const data = undoManager.data;
+      const currentLevel = Levels.createFromEditor(editor);
+
+      const dispatchEvent = (lastLevel: UndoLevel | undefined) => {
+        editor.setDirty(true);
+        editor.dispatch('change', {
+          level: currentLevel,
+          lastLevel
+        });
+      };
+
+      Arr.get(data, index.get()).fold(
```

Comment:
This logic could be simplified as we don't need the fold or separate function. eg:
```ts
const lastLevel = Arr.get(data, index.get()).getOrUndefined();
editor.setDirty(true);
editor.dispatch('change', {
  level: currentLevel,
  lastLevel
});
```

Reply:
You could even do `data[index.get()]` since we don't need to do a safe index based lookup here.

Reply:
ok, I removed all the variables since these don't give any additional information, the result is:
```JavaScript
editor.setDirty(true);
editor.dispatch('change', {
  level: Levels.createFromEditor(editor),
  lastLevel: Arr.get(undoManager.data, index.get()).getOrUndefined()
});
```